tests/TestCase/Middleware/ThrottleMiddlewareTest.php

```diff
@@ -0,0 +1,291 @@
+<?php
+namespace Muffin\Throttle\Test\TestCase\Middleware;
+
+use Cake\Cache\Cache;
+use Cake\Core\Configure;
+use Cake\Http\Response;
+use Cake\Http\ServerRequest;
+use Cake\TestSuite\TestCase;
+use Muffin\Throttle\Middleware\ThrottleMiddleware;
+use StdClass;
+
+class ThrottleMiddlewareTest extends TestCase
+{
+    /**
+     * setUp method
+     *
+     * @return void
+     */
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->skipIf(version_compare(Configure::version(), '3.4') == -1 ? true : false);
+        $this->skipIf(!function_exists('apcu_store'), 'APCu is not installed or configured properly.');
+        if ((PHP_SAPI === 'cli' || PHP_SAPI === 'phpdbg')) {
+            $this->skipIf(!ini_get('apc.enable_cli'), 'APC is not enabled for the CLI.');
+        }
+    }
+
+    /**
+     * Test __construct
+     */
+    public function testConstructor()
+    {
+        $middleware = new ThrottleMiddleware();
+        $result = $middleware->getConfig();
+
+        $this->assertEquals('Rate limit exceeded', $result['message']);
+        $this->assertEquals('+1 minute', $result['interval']);
+        $this->assertEquals(10, $result['limit']);
+        $this->assertTrue(is_callable($result['identifier']));
+
+        $expectedHeaders = [
+            'limit' => 'X-RateLimit-Limit',
+            'remaining' => 'X-RateLimit-Remaining',
+            'reset' => 'X-RateLimit-Reset'
+        ];
+        $this->assertEquals($expectedHeaders, $result['headers']);
+    }
+
+    /**
+     * Test __invoke
+     */
+    public function testInvoke()
+    {
+        Cache::drop('throttle');
+        Cache::setConfig('throttle', [
+            'className' => 'Cake\Cache\Engine\ApcEngine',
+            'prefix' => 'throttle_'
+        ]);
+
+        $middleware = new ThrottleMiddleware([
+            'limit' => 0
```

Comment:
Alright, I believe I just did a few asserts for both test cases. Let me know if I did that correctly.